src/pojo-interfaces.ts

```diff
@@ -0,0 +1,424 @@
+/* eslint-disable license-header/header */
+/**
+ * This is just a strawman proposal;
+ * the actual new interfaces will not be implemented through this file.
+ *
+ * They're also not set in stone. Actual implementation experience might result in further tweaks.
+ * Names can still be changed if we propose it as an actual specification, of course.
+ *
+ * The design goals of these interfaces are:
+ * - JSON.parse(JSON.stringify(data)) is lossless
+ * - Optimised for reading and copying in an expected shape in stateful (Solid) applications
+ * - fromRdfJsDataset(toRdfJsDataset(data)) is lossless
+ *
+ * Non-goals:
+ * - Optimising for streaming data
+ * - Optimising for unknown use cases
+ * - Be perfect from the get-go
+ * - Be immediately useful outside of solid-client
+ * @module
+ */
+
+import { dataset as rdfJsDataset } from "@rdfjs/dataset";
+import RdfJsDataFactory from "@rdfjs/data-model";
```

Comment:
I think these 2 lines might be wrong, as both are specific implementations of the RDF/JS Dataset and Data Model respectively, and not the pure interfaces. For example, I think where the Data Model is being used should be replaced by just the pure Data Model interface (i.e., you're already correctly importing `rdf-js` on the next line), and the signatures of `toRdfJsQuads()` and `toRdfJsDataset()` need to be updated to take in as a parameter the specific DataFactory instance of whatever RDF/JS implementation the developer happens to want to use.

Reply:
As mentioned in the PR description, this isn't meant to be a production implementation of translation functions, but rather just to get a feel for what working with the new types would look like, so that's why I didn't implement passing in your own DataFactory.

I did that now though: 1e2a4056

Note that the imports are still there, because of course we don't want to _require_ developers to always pass in their own; we use those implementations by default.